util/util.go

```diff
@@ -132,3 +136,26 @@ func IntV(in *int) int {
 	}
 	return *in
 }
+
+// FSGeneration generates a name and a path that will be used to feed
+// the `fsMap` in case consumer provided no entry for it in the map.
+// FSMap is required to link the filesystem from v2 to v3 with the rest of the configuration.
+func FSGeneration(name string, fsMap map[string]string) error {
+	if len(name) == 0 {
+		rand.Seed(time.Now().Unix())
+		generatedName := rand.Int()
+		name = "ignition" + strconv.Itoa(generatedName)
```

Comment:
The update of `fs.Name` is missing here. I suggest to pass `fs` to this function to be able to run `fs.Name = name` here.

Reply:
or let the function return the new `name` and the caller has to set it

Reply:
(The function parameter that gets passed is a reference and when you modify the reference to point somewhere else, the caller doesn't see this effect)